tests/gha-run-k8s-common.sh

```diff
@@ -31,12 +31,16 @@ function _print_instance_type() {
 function _print_cluster_name() {
 	local test_type="${1:-k8s}"
 	local short_sha
+	local name
 
 	if [ -n "${AKS_NAME:-}" ]; then
 		echo "$AKS_NAME"
 	else
-		short_sha="$(git rev-parse --short=12 HEAD)"
-		echo "${test_type}-${GH_PR_NUMBER}-${short_sha}-${KATA_HYPERVISOR}-${KATA_HOST_OS}-amd64-${K8S_TEST_HOST_TYPE:0:1}-${GENPOLICY_PULL_METHOD:0:1}"
+		short_sha="$(git rev-parse --short=6 HEAD)"
+		name="${test_type}-${GH_PR_NUMBER}-${short_sha}-${KATA_HYPERVISOR}-${KATA_HOST_OS}-amd64-${K8S_TEST_HOST_TYPE:0:1}-${GENPOLICY_PULL_METHOD:0:1}"
+		# An AKS cluster must have less than 64 characters. Attempt to shorten the name by removing dashes.
+		[ ${#name} -lt 64 ] || name=$(echo -n "$name" |  tr -d '-')
+		echo "$name"
```

Comment:
that might be surprising to people who assume that cluster name follow a common naming scheme, I wouldn't do that pro-actively. If we still reach > 64 chars, we have to consider moving metadata to tags

Reply:
We can't really move metadata to tags since we're using those to ensure the cluster name is unique to a given test run. Maybe we should hash it instead? Like: `${test_type}-${GH_PR_NUMBER}-HASH(${short_sha}-${KATA_HYPERVISOR}-${KATA_HOST_OS}-amd64-${K8S_TEST_HOST_TYPE:0:1}-${GENPOLICY_PULL_METHOD:0:1})`

Reply:
I thought in using hash as @sprt suggested but then the cluster name will be intelligible. Having a meaningful name is important for when there are dangling clusters and you want to back-track to the original pipeline; but given that we are already clean up all aging k8s clusters we probably won't need to inspect the clusters anymore....meaningful names become almost useless. All in all, it seems that using hashes ought to work out.

Reply:
could we atleast add the metadata to tags then? there are cases where clusters were stuck undeletable, and it helps correlated back to a GHA run.

Reply:
yes, it seems the best approach @jepio . I will re-do this PR. Thanks guys!